src/pretty.rs

```diff
@@ -0,0 +1,471 @@
+                // TODO
+                println!("pretty let pat");
+                unimplemented!()
+            }
+            App(rt1, rt2) => match rt1.as_ref() {
+                Op1(crate::term::UnaryOp::Ite(), _) => rt1
+                    .to_owned()
+                    .pretty(allocator)
+                    .append(allocator.space())
+                    .append(allocator.text("then"))
+                    .append(allocator.line())
+                    .append(rt2.to_owned().pretty(allocator).nest(2))
+                    .append(allocator.line())
+                    .append(allocator.text("else"))
+                    .group(),
+                _ => rt1
+                    .to_owned()
+                    .pretty(allocator)
+                    .append(allocator.line())
+                    .append(allocator.atom(rt2))
+                    .group(),
+            },
+            Var(id) => allocator.as_string(id),
+            Enum(id) => allocator.text(format!("`{}", id)),
+            Record(fields, attr) => allocator
+                .line()
+                .append(allocator.intersperse(
+                    fields.iter().map(|(id, rt)| {
+                        allocator
+                            .as_string(id)
+                            .append(allocator.space())
+                            .append(if let MetaValue(mv) = rt.as_ref() {
+                                allocator
+                                    .metadata(&mv, true)
+                                    .append(allocator.space())
+                                    .append(
+                                        mv.value
+                                            .clone()
+                                            .map(|v| {
+                                                allocator.text("= ").append(v.pretty(allocator))
+                                            })
+                                            .or(Some(allocator.nil()))
+                                            .unwrap(),
+                                    )
+                            } else {
+                                allocator
+                                    .text("=")
+                                    .append(allocator.line())
+                                    .append(rt.to_owned().pretty(allocator).nest(2))
+                            })
+                            .append(allocator.text(","))
+                    }),
+                    allocator.line(),
+                ))
+                .append(if attr.open {
+                    allocator.line().append(allocator.text("..."))
+                } else {
+                    allocator.nil()
+                })
+                .nest(2)
+                .append(allocator.line())
+                .group()
+                .braces(),
+            RecRecord(
+                fields,
+                inter_fields, /* field whose name is defined by interpolation */
```

Comment:
it's a detail, but `inter_fields` is called `dyn_fields` elsewhere in the codebase for dynamically defined fields. Unless you specifically care that this one be called `inter_fields`, it's nice to keep naming consistent.